programs/svm-spoke/src/state/refund_account.rs

```diff
@@ -0,0 +1,132 @@
+    pub fn try_from_remaining_account(
+        remaining_accounts: &'c [AccountInfo<'info>],
+        index: usize,
+        expected_token_account: &Pubkey,
+        expected_mint: &Pubkey,
+        token_program: &Pubkey,
+    ) -> Result<Self> {
+        let refund_account_info = remaining_accounts
+            .get(index)
+            .ok_or(ErrorCode::AccountNotEnoughKeys)?;
+
+        Self::try_token_account_from_account_info(
+            refund_account_info,
+            expected_token_account,
+            expected_mint,
+            token_program,
+        )
+        .map(Self::TokenAccount)
+        .or_else(|| {
+            Self::try_claim_account_from_account_info(
+                refund_account_info,
+                expected_mint,
+                expected_token_account,
+            )
+            .map(Self::ClaimAccount)
+        })
+        .ok_or_else(|| {
+            error::Error::from(CustomError::InvalidRefund)
+                .with_account_name(&format!("remaining_accounts[{}]", index))
+        })
+    }
+
+    // This implements the following Anchor account constraints when parsing remaining account as a token account:
+    // #[account(
+    //     mut,
+    //     address = expected_token_account @ CustomError::InvalidRefund,
+    //     token::mint = expected_mint,
+    //     token::token_program = token_program
+    // )]
+    // pub token_account: InterfaceAccount<'info, TokenAccount>,
+    // Note: All errors are ignored and Option is returned as we do not log them anyway due to memory constraints.
+    fn try_token_account_from_account_info(
+        account_info: &'info AccountInfo<'info>,
+        expected_token_account: &Pubkey,
+        expected_mint: &Pubkey,
+        token_program: &Pubkey,
+    ) -> Option<InterfaceAccount<'info, TokenAccount>> {
+        // Checks ownership on deserialization for the TokenAccount interface.
+        let token_account: InterfaceAccount<'info, TokenAccount> =
+            InterfaceAccount::try_from(account_info).ok()?;
+
+        // Checks if the token account is writable.
+        if !account_info.is_writable {
+            return None;
+        }
+
+        // Checks the token address matches.
+        if account_info.key != expected_token_account {
+            return None;
+        }
+
+        // Checks if the token account is associated with the expected mint.
+        if &token_account.mint != expected_mint {
+            return None;
+        }
+
+        // Checks ownership by specific token program.
+        if account_info.owner != token_program {
+            return None;
+        }
+
+        Some(token_account)
+    }
+
+    // This implements the following Anchor account constraints when parsing remaining account as a claim account:
+    // #[account(
+    //     mut,
+    //     seeds = [b"claim_account", mint.key().as_ref(), token_account.key().as_ref()],
+    //     bump
+    // )]
+    // pub claim_account: Account<'info, ClaimAccount>,
```

Comment:
is this meant to be commented out?

Reply:
Yes, it is meant to document what Anchor macros this method is reimplementing, so its easier to audit knowing the intention of the implementation.